Fix issues using Hydra and Storm with OpenGL core profile

brechtvl opened the pull request:
### Description of Change(s)

These changes together make Hydra and Storm work without warnings in Blender, which uses the OpenGL core profile rather than the compatibility profile.

* Don't use point size or line width for core profile, since these are not available and generate many warnings. This only silences the warnings, ideally these features would be implemented in shaders.
* Fix missing VAO binding that is required with the core profile, causing Storm renders to fail.
* Also fix VAO double free and leak in GL graphics pipeline that I noticed looking into this.

---

The solution for VAOs in `HdSt_IndirectDrawBatch` does not seem ideal, as OpenGL usage should be abstracted away in `hgiGL`. However OpenGL vertex buffer binding still seems to be part of `hdSt` and it's not clear to me where it will move exactly.

Also note that while it may seem that in `HdSt_IndirectDrawBatch::_ExecuteFrustumCull`, `cullGfxCmds->BindPipeline` will bind the VAO, this doesn't actually work because those commands are queued and only executed after functions like `state.BindResourcesForViewTransformation` that bind the vertex buffer objects.

Advice for a better solution would be welcome.

- [x] I have verified that all unit tests pass with the proposed changes
- [x] I have submitted a signed Contributor License Agreement

brechtvl committed:
Fix invalid enum errors with OpenGL core profile and Storm

Point smooth and line width are only available with the compatibility
profile. NVIDIA drivers report errors when trying to use these with the
core profile.

brechtvl committed:
Fix Storm and OpenGL core profile failing to render due missing VAO

In compatibility profile this is not required, but for core profile a
VAO is always needed when binding VBOs.

brechtvl committed:
Fix VAO double free and leak in GL graphics pipeline

jesschimein commented:
Filed as internal issue #USD-8536

brechtvl committed:
Merge branch 'release' into fix-core-profile

pixar-oss committed:
Another simpleShading tutorial fix

(Internal change: 2353635)

anwang2009 committed:
Report 25.02 performance test results for open source USD assets

(Internal change: 2353764)

pixar-oss committed:
Merge release v25.02

katpet committed:
Fix one reference to mx::Type::VECTOR2 which changed to not being a pointer in MaterialX 1.39

matthewcpp committed:
This change  addresses a crash that resulted when adding / removing a sublayer results in a cycle.  In order to prevent this, a cache of visited layer/sublayer path pairs is used to terminate recursive processing of sublayer paths.

Fixes #3493

(Internal change: 2355098)

mhamid3d committed:
Add ZLIB to requiredDependencies if HDF5 is enabled

Alembic requires ZLIB, but only in the case that HDF5 support is enabled.
Since we are selectively adding zlib as a dependency, we need to append
zlib to `requiredDependencies` if HDF5 is enabled in the build context.
This fixes alembic build failures on Windows.

Closes #3501

(Internal change: 2355438)

matthewcpp committed:
This change fixes a coding error that is triggered when performing a sublayer operation on a layer whose file format is a package.  The current implementation which generates fine grained change lists makes use of an anonymous layer to compute a diff against.  It is an error to create such a layer with a package file format.  To workaround this limitation with the current implementation, we send packages down the "big bang" invalidation path.

(Internal change: 2355629)

sunyab committed:
CHANGELOG: Update for 25.02a release

(Internal change: 2355759)

sunyab committed:
Update BuildUSD action to use upload-artifacts@v4 on Windows

upload-artifacts@v3 had previously been deprecated and was
disabled as of January 30th per:

https://github.blog/changelog/2024-04-16-deprecation-notice-v3-of-the-artifact-actions/

The bug that prevented us from using v4 earlier on Windows
appears to have been fixed.

(Internal change: 2355870)

DarkDefender commented:
@unhyperbolic Is there any update on this? Sorry if I'm poking the wrong person.
We are applying multiple patches to the USD version that we officially ship with Blender, this is one of them.
As we are up to quite a few patches at this point (7) and more are starting to pile up, we would like to have these fixes upstream.

jesschimein commented:
/AzurePipelines run

brechtvl committed:
Merge branch 'dev' into fix-core-profile

jesschimein commented:
/AzurePipelines run

jesschimein commented:
Hi @DarkDefender we're hoping to take a look at this PR soon!

pixar-oss committed:
Merge release v25.05

pixar-oss committed:
Fixing typo in 25.05 release notes

(Internal change: 2365654)

o2b3d committed:
Fix a crash when sampling float or half valued splines.

There was a code path that failed to set _times and _knots in
_Sampler::_UnrollInnerLoops that has been fixed and the tests have been
improved to test those code paths.

It turns out that all of our testing was using double valued splines,
so TsTest_TsEvaluator has been taught to produce splines with float and
half values. This highlighted another numeric issue with converting
extreme tangent slopes and code was added to generate reasonable
tangents, even with a reduced value range.

Fixes #3614

(Internal change: 2365974)

matthewcpp committed:
This reverts a change to PCP introduced in 2362362.  While the change to PcpChanges::DidChangeSpecs is required for finer grained change notifications for sublayer operations, it introduces a bug in the USD layer when taken in isolation without forthcoming changes to USD change processing.  This change will be re-introduced in the future along with the required updates to USD change processing.

A test which exercises the incorrect behavior introduced by the (now reverted) code has also been added.

Fixes #3615

(Internal change: 2366475)

meshula committed:
Fix matrix mult order for GfColor constructor

(Internal change: 2367351)

sunyab committed:
CHANGELOG: Update for 25.05.01 release

(Internal change: 2367452)

pixar-oss committed:
Merge release v25.08

brechtvl committed:
Merge tag 'v25.08' into fix-core-profile

brechtvl committed:
Merge branch 'dev' into fix-core-profile
